Dockerfile.f31

```diff
@@ -0,0 +1,111 @@
+# Build
+# -----
+# $ podman build --build-arg TYPE=local -t dnf-bot/ci-dnf-stack:f31 -f Dockerfile.f31
+#
+#
+# Run
+# ---
+# $ podman run -it dnf-bot/ci-dnf-stack:f31 behave -Ddnf_executable=dnf --junit --junit-directory=/opt/behave/junit/ [--wip --no-skipped]
+#
+#
+# Build types
+# -----------
+# distro
+#       install distro packages
+# copr
+#       install distro packages
+#       then upgrade to copr packages
+# local
+#       install distro packages
+#       then upgrade to copr packages
+#       then install packages from local rpms/ folder
+#       install also additional tools for debugging in the container
+
+
+FROM fedora:31
+ENV LANG C
+ARG TYPE=local
+ARG OSVERSION=fedora__31
+
+
+
+# disable deltas and weak deps
+RUN set -x && \
+    echo -e "deltarpm=0" >> /etc/dnf/dnf.conf && \
+    echo -e "install_weak_deps=0" >> /etc/dnf/dnf.conf
+
+
+# if TYPE == copr or local, enable nightly copr
+RUN set -x && \
+    if [ "$TYPE" == "copr" -o "$TYPE" == "local" ]; then \
+        dnf -y install dnf-plugins-core; \
+        dnf -y copr enable rpmsoftwaremanagement/dnf-nightly; \
+    fi
+
+
+# upgrade all packages to the latest available versions
+RUN set -x && \
+    dnf -y --refresh upgrade
+
+
+# install the test environment and additional packages
+RUN set -x && \
+    dnf -y install \
+        # behave and test requirements
+        findutils \
+        libfaketime \
+        python3-behave \
+        python3-pexpect \
+        rpm-build \
+        rpm-sign \
+        # if TYPE == local, install debugging tools
+        $(if [ "$TYPE" == "local" ]; then \
+            echo \
+                less \
+                openssh-clients \
+                procps-ng \
+                psmisc \
+                screen \
+                strace \
+                tcpdump \
+                vim-enhanced \
+                vim-minimal \
+                wget \
+            ; \
+        fi) \
+        # install dnf stack
+        createrepo_c \
+        dnf \
+        yum \
+        dnf-plugins-core \
+        dnf-utils \
+        # all plugins with the same version as dnf-utils
+        $(dnf repoquery dnf-utils --latest-limit=1 -q --qf="python*-dnf-plugin-*-%{version}-%{release}") \
+        libdnf \
+        microdnf
+
+
+# install local RPMs if available
+COPY ./rpms/ /opt/behave/rpms/
+RUN rm /opt/behave/rpms/*-{devel,debuginfo,debugsource}*.rpm; \
+    if [ -n "$(find /opt/behave/rpms/ -maxdepth 1 -name '*.rpm' -print -quit)" ]; then \
+        dnf -y install /opt/behave/rpms/*.rpm --disableplugin=local; \
+    fi
+
+
+# copy test suite
+COPY ./dnf-behave-tests/ /opt/behave/
+
+# set os userdata for behave
+RUN echo -e "\
+[behave.userdata]\n\
+os=$OSVERSION" > /opt/behave/behave.ini
+
+# build test repos from sources
+RUN set -x && \
+    cd /opt/behave/fixtures/specs/ && \
+    ./build.sh --force-rebuild
+
+
+VOLUME ["/opt/behave/junit"]
+WORKDIR /opt/behave
```

Comment:
Could the new Dockerfile go into a separate commit?

Reply:
Done.